src/Filter/RatingBasedManifestFilter/RatingBasedManifestsFilter.php

```diff
@@ -0,0 +1,167 @@
+        ConfigurationProviderInterface $configurationProvider,
+        ModuleRatingFetcherInterface $moduleRatingFetcher,
+        ManifestToModulesRatingRequestMapper $manifestToModulesRatingRequestMapper
+    ) {
+        $this->configurationProvider = $configurationProvider;
+        $this->moduleRatingFetcher = $moduleRatingFetcher;
+        $this->manifestToModulesRatingRequestMapper = $manifestToModulesRatingRequestMapper;
+    }
+
+    /**
+     * @param array<mixed> $manifests
+     *
+     * @return array<mixed>
+     */
+    public function filterManifests(array $manifests): array
+    {
+        $requiredRating = $this->configurationProvider->getManifestsRatingThreshold();
+
+        if ($requiredRating <= static::MIN_RATING_THRESHOLD) {
+            return $manifests;
+        }
+
+        if ($requiredRating > static::MAX_RATING_THRESHOLD) {
+            return [];
+        }
+
+        $modulesRatingResponse = $this->moduleRatingFetcher->fetchModulesRating(
+            $this->manifestToModulesRatingRequestMapper->mapManifestsToModulesRatingRequest($manifests),
+        );
+
+        $filteredManifests = $this->filterManifestsByModulesRating(
+            $manifests,
+            $this->indexResponseByModuleIdWithRating($modulesRatingResponse),
+            $requiredRating,
+        );
+
+        return $this->removeEmptyModules($filteredManifests);
+    }
+
+    /**
+     * @param array<mixed> $manifests
+     * @param array<string, int> $modulesRatings
+     * @param int $requiredRating
+     *
+     * @return array<mixed>
+     */
+    protected function filterManifestsByModulesRating(
+        array $manifests,
+        array $modulesRatings,
+        int $requiredRating
+    ): array {
+        foreach ($manifests as $fullModuleName => $strategies) {
+            foreach ($strategies as $strategyName => $strategy) {
+                foreach ($strategy as $index => $manifest) {
+                    [$organization, $moduleName] = explode('.', $manifest[IntegratorConfig::MODULE_KEY]);
+                    $moduleVersionName = $manifest[IntegratorConfig::MODULE_VERSION_KEY];
+
+                    $moduleId = sprintf('%s:%s:%s', $organization, $moduleName, $moduleVersionName);
+
+                    if (!isset($modulesRatings[$moduleId]) || $modulesRatings[$moduleId] < $requiredRating) {
+                        unset($manifests[$fullModuleName][$strategyName][$index]);
+                    }
+                }
+            }
+        }
+
+        return $manifests;
+    }
+
+    /**
+     * @param \SprykerSdk\Integrator\Communication\ReleaseApp\ModulesRatingResponseDto $modulesRatingResponseDto
+     *
+     * @return array<string, int>
+     */
+    protected function indexResponseByModuleIdWithRating(ModulesRatingResponseDto $modulesRatingResponseDto): array
+    {
+        $indexedModuleRating = [];
+
+        foreach ($modulesRatingResponseDto->getModuleRatingResponseDtos() as $moduleRatingResponseDto) {
```

Comment:
I've checked response mapping in the release app branch and found the following logic.
```
            $moduleId = sprintf('%s:%s:%s', $organizationName, $module['name'], $module['version']);
```
is it possible to reuse the value of the array key?